MonoGame.Framework/Graphics/Vertices/VertexBuffer.cs

```diff
@@ -26,10 +26,6 @@ protected VertexBuffer(GraphicsDevice graphicsDevice, VertexDeclaration vertexDe
             this.VertexCount = vertexCount;
             this.BufferUsage = bufferUsage;
 
-            // Make sure the graphics device is assigned in the vertex declaration.
-            if (vertexDeclaration.GraphicsDevice != graphicsDevice)
-                vertexDeclaration.GraphicsDevice = graphicsDevice;

```

Comment:
Those lines were restored in the last commit. Now that GraphicsDevice properly disposes of everything, it is not an issue.